pkg/monitortests/testframework/etcddiskmetricsintervals/monitortest.go

```diff
@@ -0,0 +1,229 @@
+package etcddiskmetricsintervals
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/library-go/test/library/metrics"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/monitortestlibrary/prometheus"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	prometheustypes "github.com/prometheus/common/model"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type etcdDiskMetricsCollector struct {
+	adminRESTConfig         *rest.Config
+	commitDurationThreshold float64
+	walFsyncThreshold       float64
+}
+
+func NewEtcdDiskMetricsCollector() monitortestframework.MonitorTest {
+	return &etcdDiskMetricsCollector{
+		commitDurationThreshold: 0.025, // 25ms threshold, defined upstream
+		walFsyncThreshold:       0.01,  // 10ms threshold, defined upstream
+	}
+}
+
+func (w *etcdDiskMetricsCollector) PrepareCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	return nil
+}
+
+func (w *etcdDiskMetricsCollector) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	w.adminRESTConfig = adminRESTConfig
+	return nil
+}
+
+func (w *etcdDiskMetricsCollector) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	logger := logrus.WithField("MonitorTest", "EtcdDiskMetricsCollector")
+
+	intervals, err := w.buildIntervalsForEtcdDiskMetrics(ctx, w.adminRESTConfig, beginning)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	logger.Infof("collected %d etcd disk metrics intervals", len(intervals))
+	return intervals, nil, nil
+}
+
+func (w *etcdDiskMetricsCollector) buildIntervalsForEtcdDiskMetrics(ctx context.Context, restConfig *rest.Config, startTime time.Time) ([]monitorapi.Interval, error) {
+	logger := logrus.WithField("func", "buildIntervalsForEtcdDiskMetrics")
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	routeClient, err := routeclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = kubeClient.CoreV1().Namespaces().Get(ctx, "openshift-monitoring", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return []monitorapi.Interval{}, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	prometheusClient, err := metrics.NewPrometheusClient(ctx, kubeClient, routeClient)
+	if err != nil {
+		return nil, err
+	}
+
+	if intervals, err := prometheus.EnsureThanosQueriersConnectedToPromSidecars(ctx, prometheusClient); err != nil {
```

Comment:
I would change intervals to _ here and explicitly return nil for the error case.  Just for clarity (AI Slop I'm sure)